mono/metadata/object.c

```diff
@@ -6415,7 +6415,7 @@ mono_string_empty_handle (MonoDomain *domain)
  * \returns A newly created string object which contains \p text.
  */
 MonoString *
-mono_string_new_utf16 (MonoDomain *domain, const guint16 *text, gint32 len)
+mono_string_new_utf16 (MonoDomain *domain, const gunichar2 *text, gint32 len)
```

Comment:
This is declared in object.h as
```c
MONO_API MONO_RT_EXTERNAL_ONLY
MonoString*
mono_string_new_utf16	    (MonoDomain *domain, const mono_unichar2 *text, int32_t len);
```

and `mono_unichar2` is guint16 (see `mono-publib.h`)

Reply:
Either way ok with me. 'g' is brief, `mono` matches, and must match.

This is all a breaking change, btw, however pretty rare, and the C++ compiler broke people the same way and more.

i.e.
```
MonoString (*function_pointer)(MonoDomain*, const guint16*, gint32 len) = mono_string_new_utf16;
```

must be changed to:
```MonoString (*function_pointer)(MonoDomain*, const gunichar2*, gint32 len) = mono_string_new_utf16;
```

The larger C++ breakage was the name mangling level.
The break is embedders, using C++, on Windows, taking address, w/o templatizing or casting.

Reply:
Similarly:

```
MONO_API mono_unichar2 *mono_string_chars  (MonoString *s);
```

```
const gunichar2 *p = mono_string_chars (s);
```

Where is the boundary that we transition to `g`?